agent/queryparser/parser.go

```diff
@@ -39,3 +39,36 @@ func MySQL(q string) (string, uint32, error) {
 
 	return parsedQuery.Query, uint32(len(bindVars)), nil
 }
+
+// MySQLComments parse query and return its comments. Can parse multi comments.
+// Multi comments support should be dropped in future MySQL versions.
+// Doc: https://dev.mysql.com/doc/refman/8.0/en/comments.html
+func MySQLComments(q string) ([]string, error) {
+	comments, err := parseMySQLComments(q)
+	if err != nil {
+		return nil, err
+	}
+
+	var res []string //nolint:prealloc
+	for c := range comments {
+		res = append(res, c)
+	}
+
+	return res, nil
+}
+
+// PostgreSQLComments parse query and return its comments. Can parse multi comments.
+// Doc: https://www.postgresql.org/docs/current/sql-syntax-lexical.html#SQL-SYNTAX-COMMENTS
+func PostgreSQLComments(q string) ([]string, error) {
+	comments, err := parsePostgreSQLComments(q)
+	if err != nil {
+		return nil, err
+	}
+
+	var res []string //nolint:prealloc
```

Comment:
Any reason we don't allocate the slice on initialization?

Reply:
I rather doing this when its really needed, but it you want I can change it for sure.